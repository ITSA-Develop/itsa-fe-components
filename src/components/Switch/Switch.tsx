import { Switch as AntSwitch, SwitchProps } from 'antd';

export const Switch = ({ ...rest }: SwitchProps) => {
	return <AntSwitch {...rest} />;
};
