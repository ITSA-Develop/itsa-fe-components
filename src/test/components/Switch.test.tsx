import { fireEvent, render, screen } from '@testing-library/react';
import { Switch } from '../../components/Switch/Switch';

describe('Switch component', () => {
	it('renders correctly and toggles', () => {
		const onChange = vi.fn();
		const { container } = render(<Switch onChange={onChange} />);

		const switchElement = screen.getByRole('switch');
		expect(switchElement).toBeInTheDocument();

		fireEvent.click(switchElement);
		expect(onChange).toHaveBeenCalled();
		expect(container).toMatchSnapshot();
	});
});
