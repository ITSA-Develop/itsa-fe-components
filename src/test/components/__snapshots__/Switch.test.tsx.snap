// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch component > renders correctly and toggles 1`] = `
<div>
  <button
    aria-checked="true"
    class="ant-switch css-dev-only-do-not-override-l9pxc0 ant-switch-checked"
    role="switch"
    type="button"
  >
    <div
      class="ant-switch-handle"
    />
    <span
      class="ant-switch-inner"
    >
      <span
        class="ant-switch-inner-checked"
      />
      <span
        class="ant-switch-inner-unchecked"
      />
    </span>
  </button>
</div>
`;
